src/hid/audio.h

```diff
@@ -23,15 +23,23 @@ class AudioHandle
         ERR,
     };
 
+    typedef const float* const* InputBuffer;
+    typedef float**             OutputBuffer;
+
     /** Non-Interleaving Callback format. Both arrays arranged by float[chn][sample] */
-    typedef void (*AudioCallback)(float** in, float** out, size_t size);
+    typedef void (*AudioCallback)(InputBuffer  in,
+                                  OutputBuffer out,
+                                  size_t       size);
+
+    typedef const float* InterleavingInputBuffer;
+    typedef float*       InterleavingOutputBuffer;
 
     /** Non-Interleaving Callback format. 
      ** audio is prepared as { L0, R0, L1, R1, . . . LN, RN }
      */
-    typedef void (*InterleavingAudioCallback)(float* in,
-                                              float* out,
-                                              size_t size);
+    typedef void (*InterleavingAudioCallback)(InterleavingInputBuffer  in,
+                                              InterleavingOutputBuffer out,
+                                              size_t                   size);
```

Comment:
I suggest we move the format documentation to the buffers themselves, and then add a separate bit of documentation for the Callbacks (that doesn't include the format information).

That'll help make it a bit easier to see what's what with editors using intellisense, and when browsing the generated documentation.